docs/docs/typescript/strict-mode.md

```diff
@@ -0,0 +1,239 @@
+```
+
+You'll have to adjust the generated code depending on your User model.
+
+<details>
+<summary>Example code diffs</summary>
+
+#### A. If your project does not use roles
+
+If your `getCurrentUser` doesn't return `roles`, and you don't use this functionality, you can safely remove the `hasRoles` function.
+
+#### B. Roles on current user is a string
+
+Alternatively, if  you define the roles as a string, you can remove the code that does checks against Arrays
+
+```diff title="api/src/lib/auth.ts"
+export const hasRole = (roles: AllowedRoles): boolean => {
+  if (!isAuthenticated()) {
+    return false
+  }
+
+  const currentUserRoles = context.currentUser?.roles
+
+  if (typeof roles === 'string') {
+-    if (typeof currentUserRoles === 'string') {
+      return currentUserRoles === roles
+-    }
+  }
+
+  if (Array.isArray(roles)) {
+-    if (Array.isArray(currentUserRoles)) {
+-      return currentUserRoles?.some((allowedRole) =>
+-        roles.includes(allowedRole)
+-      )
+-    } else if (typeof context?.currentUser?.roles === 'string') {
+      // roles to check is an array, currentUser.roles is a string
+      return roles.some(
+        (allowedRole) => context.currentUser?.roles === allowedRole
+      )
+-    }
+  }
+
+  // roles not found
+  return false
+}
+```
+
+#### C. Roles on current user is an Array of strings
+
+If in your User model, roles are an array of strings, and can never be just a string, you can safely remove most of the code
+
+```diff title="api/src/lib/auth.ts"
+export const hasRole = (roles: AllowedRoles): boolean => {
+  if (!isAuthenticated()) {
+    return false
+  }
+
+ const currentUserRoles = context.currentUser?.roles
+
+  if (typeof roles === 'string') {
+-    if (typeof currentUserRoles === 'string') {
+-      return currentUserRoles === roles
+-    } else if (Array.isArray(currentUserRoles)) {
+      // roles to check is a string, currentUser.roles is an array
+      return currentUserRoles?.some((allowedRole) => roles === allowedRole)
+-    }
+  }
+
+  if (Array.isArray(roles)) {
+-    if (Array.isArray(currentUserRoles)) {
+      return currentUserRoles?.some((allowedRole) =>
+        roles.includes(allowedRole)
+      )
+-    } else if (typeof currentUserRoles === 'string') {
+-      return roles.some(
+-        (allowedRole) => currentUserRoles === allowedRole
+-      )
+    }
+  }
+
+  // roles not found
+  return false
+}
+```
+</details>
+
+### `getCurrentUser` in `api/src/lib/auth.ts`
+
+Depending on your auth provider—i.e., anything but dbAuth—because it could change based on your account settings (if you include roles or other metadata), we can't know the shape of your decoded token at setup time.
+So you'll have to make sure that the `getCurrentUser` function is typed.
+
+To help you get started, the comments above the `getCurrentUser` function describe its parameters' types. We recommend typing `decoded` without using imported types from Redwood, as this may be a little too generic!
```

Comment:
I'm a little confused here cause it seems like we say not to use imported types from Redwood, but one of the examples uses an imported type from Redwood?

Reply:
Yeah I included both, to show how you can achieve it both ways. While importing the type "works" - its not terribly useful in actual use, which is why I'm suggesting to just type it yourself